docs/source/mpi4py.util.pool.rst

```diff
@@ -0,0 +1,75 @@
+mpi4py.util.pool
+----------------
+
+.. module:: mpi4py.util.pool
+   :synopsis: :mod:`multiprocessing.pool` interface via :mod:`mpi4py.futures`.
+
+.. versionadded:: 4.0.0
+
+.. seealso::
+
+   This module intends to be a drop-in replacement for the
+   :mod:`multiprocessing.pool` interface from the Python standard library.
+   The :class:`~mpi4py.util.pool.Pool` class exposed here is implemented as a
+   thin wrapper around :class:`~mpi4py.futures.MPIPoolExecutor`.
+
+.. note::
+
+   The :mod:`mpi4py.futures` package offers a higher level interface
+   for asynchronously pushing tasks to MPI worker process, allowing
+   for a clear separation between submitting tasks and waiting for the
+   results.
+
+
+.. autoclass:: mpi4py.util.pool.Pool
+
+   .. automethod:: __init__
+
+   .. automethod:: apply
+
+   .. automethod:: apply_async
+
+   .. automethod:: map
+
+   .. automethod:: imap
+
+   .. automethod:: imap_unordered
+
+   .. automethod:: map_async
```

Comment:
nit: follow the same order as in Python docs
```suggestion
   .. automethod:: map_async

   .. automethod:: map

   .. automethod:: imap_unordered
```

Reply:
The order here in docs is the same as the order the methods are defined in the source code. 
Right now the order is `foo`, `ifoo[_unordered]`, `foo_async`.
You are proposing a swappint to `foo`, `foo_async`, `ifoo[_unordered]`.
I think it is ultimately a matter of taste, but perhaps your way is easier to follow / reason about.

Reply:
I've reordered things as per your suggestion.